features/kafka/producer/src/main/java/org/opennms/features/kafka/producer/ProtobufMapper.java

```diff
@@ -87,17 +87,17 @@ public ProtobufMapper(EventConfDao eventConfDao, HwEntityDao hwEntityDao, Sessio
                 public OpennmsModelProtos.NodeCriteria load(Long nodeId)  {
                     return sessionUtils.withReadOnlyTransaction(() -> {
                         final OnmsNode node = nodeDao.get(nodeId.intValue());
-                        if (node != null && node.getForeignId() != null && node.getForeignSource() != null) {
-                            return OpennmsModelProtos.NodeCriteria.newBuilder()
-                                    .setId(nodeId)
-                                    .setForeignId(node.getForeignId())
-                                    .setForeignSource(node.getForeignSource())
-                                    .build();
-                        } else {
-                            return OpennmsModelProtos.NodeCriteria.newBuilder()
-                                    .setId(nodeId)
-                                    .build();
+                        OpennmsModelProtos.NodeCriteria.Builder builder = OpennmsModelProtos.NodeCriteria.newBuilder();
+                        if (node == null) {
+                            return builder.setId(nodeId).build();
```

Comment:
`nodeId` should always be set to remain backwards-compatible, should it not?  Here you're only setting it if `node == null` but the other codepath never gets it.

Reply:
Oh yeah, good catch !